motoman_sia50_support/urdf/sia50d_macro.xacro

```diff
@@ -0,0 +1,209 @@
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_4_u.stl"/>
+                </geometry>
+                <material name="blue"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_4_u.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_r">
+            <visual>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_5_r.stl"/>
+                </geometry>
+                <material name="silver"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_5_r.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_b">
+            <visual>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_6_b.stl"/>
+                </geometry>
+                <material name="blue"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_6_b.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_t">
+            <visual>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_7_t.stl" />
+                </geometry>
+                <material name="silver"/>
+            </visual>
+            <collision>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_7_t.stl" />
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}tool0" />
+        <!-- end of link list -->
+        <!-- joint list -->
+        <joint name="${prefix}joint_s" type="revolute">
+            <parent link="${prefix}base_link"/>
+            <child link="${prefix}link_s"/>
+            <origin xyz="0 0 0.312000" rpy="0 0 0"/>
+            <axis xyz="0 0 1" />
+            <!-- Max Acc/Dec Torque: 4903.32, Rated Motor Torque: 18.5 -->
+            <limit lower="-3.1415" upper="3.1415" effort="18.5" velocity="2.967051" />
```

Comment:
Please use the `radians(..)` conversion factor and specify limits in degrees (so they can be easily compared to the specs).

Additionally: `18.5` Nm for the first joint? This seems rather low.

If you want to include an `effort` limit here, you should probably list the other number (ie: maximum acc/dec torque).